docs/src/references/c8y-configuration-management.md

```diff
@@ -447,64 +447,35 @@ and of HTTP to let the child device `PUT` the requested file.
-      or `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/c8y-configuration-plugin`
-      (depending on the file has been pushed by the cloud or the child device).
-   1. The `c8y_configuration_plugin` puts this copy under
-      `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`,
-      and uses it for an updated list of configuration files
-      (to which this file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`
-      is implicitly added with type `c8y-configuration-plugin`).
-   1. If the request were triggered by the cloud,
-      the `c8y_configuration_plugin` proceeds on processing the `successful` status message,
-      notifying the cloud and removing the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
-      Note that the request might have been triggered by the main device on start,
-      or by the child device itself (see below),
-      in that case the cloud must not be notified. 
-1. On start, the `c8y_configuration_plugin` check for each child device `$CHILD_DEVICE_ID`,
-   listed in the directory `$TEDGE_CONFIG_DIR/c8y/`,
-   if there is a known `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
-   1. If the configuration is missing for some `$CHILD_DEVICE_ID`
-      (i.e. if the directory `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID` exists
-      but not the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`).
-   1. Then the `c8y_configuration_plugin` request this file from the child device
-      by sending an MQTT message on the topic `tedge/$CHILD_DEVICE_ID/commands/req/config_snapshot`.
-   1. The payload of this message is a JSON record with 2 required fields:
-       * `"url": "http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_snapshot/c8y-configuration-plugin"`
-       * `"type": "c8y-configuration-plugin"`    
-1. The child-device configuration software has nothing specific to do on requests
-   that are related to the `"type": "c8y-configuration-plugin"`.
-   However, on start and on local changes of the list of configuration files,
-   The child-device configuration software should push to the thin-edge device
-   this new or updated content. This is done as a regular snapshot request.
-      1. `PUT`s the content of updated `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`
-         to the `http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_snapshot/c8y-configuration-plugin`.
-      1. On success of the upload, the child-device configuration software
+The configuration files for a child device `$CHILD_DEVICE_ID` are defined in
+the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` on the thin-edge device.
+But, the content of this file must be provided by the child-device agent
+by uploading a `c8y-configuration-plugin.toml` with `"type": "c8y-configuration-plugin"` in its prescribed format.
+
+The child-device can upload this file on its startup,
```

Comment:
If the child-device fails to send upload its own configuration then configuration management will never be enabled for it.  Keep the `--init $CHILD` option, as a work around, would not be enough since the the child-device configuration file list must be sent in any case.

```suggestion
The child-device must upload this file on its startup as well as when updated,
```

Reply:
With that *can* instead of *must* I was trying to stress that the child-device agent can upload this config list **at any point** which **need not necessarily be during the startup** (although startup is one logical place to do so). But now I see how it might be interpreted as an optional step by a reader. Sending this list at some point is a must. So, I'll update that statement as you suggested.